Libplanet.Tests/Blockchain/BlockChainTest.ValidateNextBlock.cs

```diff
@@ -171,23 +171,80 @@ public void ValidateNextBlockInvalidTimestamp()
             var policy = new BlockPolicy(
                 blockInterval: TimeSpan.FromMilliseconds(3 * 60 * 60 * 1000)
             );
-            var stateStore = new TrieStateStore(stateKeyValueStore);
-            IStore store = new MemoryStore();
-            var actionEvaluator = new ActionEvaluator(
+            var stateStore1 = new TrieStateStore(new MemoryKeyValueStore());
+            IStore store1 = new MemoryStore();
+            var actionEvaluator1 = new ActionEvaluator(
                 _ => policy.BlockAction,
-                stateStore,
+                stateStore1,
                 new SingleActionLoader(typeof(DumbAction)));
             var genesisBlock = TestUtils.ProposeGenesisBlock(
-                actionEvaluator,
                 TestUtils.ProposeGenesis(TestUtils.GenesisProposer.PublicKey),
                 TestUtils.GenesisProposer);
-            store.PutBlock(genesisBlock);
-            Assert.NotNull(store.GetStateRootHash(genesisBlock.Hash));
+            var chain1 = BlockChain.Create(
+                policy,
+                new VolatileStagePolicy(),
+                store1,
+                stateStore1,
+                genesisBlock,
+                actionEvaluator1);
+
+            var policyWithBlockAction = new BlockPolicy(
+                new SetStatesAtBlock(default, (Text)"foo", default, 0),
+                policy.BlockInterval
+            );
+            var stateStore2 = new TrieStateStore(new MemoryKeyValueStore());
+            IStore store2 = new MemoryStore();
+            var actionEvaluator2 = new ActionEvaluator(
+                _ => policyWithBlockAction.BlockAction,
+                stateStore2,
+                new SingleActionLoader(typeof(DumbAction)));
+            var chain2 = BlockChain.Create(
+                policyWithBlockAction,
+                new VolatileStagePolicy(),
+                store2,
+                stateStore2,
+                genesisBlock,
+                actionEvaluator2);
 
+            Block block1 = chain1.EvaluateAndSign(
+                new BlockContent(
+                    new BlockMetadata(
+                        protocolVersion: BlockMetadata.CurrentProtocolVersion,
+                        index: 1,
+                        timestamp: genesisBlock.Timestamp.AddSeconds(1),
+                        miner: TestUtils.GenesisProposer.Address,
+                        publicKey: TestUtils.GenesisProposer.PublicKey,
+                        previousHash: genesisBlock.Hash,
+                        txHash: null,
+                        lastCommit: null)).Propose(),
+                TestUtils.GenesisProposer);
+
+            Assert.Throws<InvalidBlockStateRootHashException>(() =>
+                chain2.Append(block1, TestUtils.CreateBlockCommit(block1)));
+
+            chain1.Append(block1, TestUtils.CreateBlockCommit(block1));
+        }
+
+        [Fact]
+        public void ValidateNextBlockInvalidStateRootHashBeforePostpone()
```

Comment:
New test for validate next block - when both of previous block and appending block are before-sloth